src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,1009 @@
+from binascii import a2b_hex
+from logging import exception
+import os
+import yaml
+import json
+import datetime
+from subprocess import Popen, PIPE, run, STDOUT, call, DEVNULL
+import shutil
+from knack.log import get_logger
+from azure.cli.core import telemetry
+import azext_connectedk8s._constants as consts
+logger = get_logger(__name__)
+# pylint:disable=unused-argument
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=line-too-long
+
+
+diagnoser_output = []
+
+
+def create_folder_diagnosticlogs(time_stamp):
+
+    global diagnoser_output
+
+    try:
+
+        # Fetching path to current directory to create the arc diagnostic folder
+        home_dir = os.path.expanduser('~')
+        filepath = os.path.join(home_dir, '.azure', 'arc_diagnostic_logs')
+
+        # Creating Diagnostic folder and its subfolder with the given timestamp and cluster name to store all the logs
+        try:
+            os.mkdir(filepath)
+        except FileExistsError:
+            pass
+
+        filepath_with_timestamp = os.path.join(filepath, time_stamp)
+        try:
+            os.mkdir(filepath_with_timestamp)
+        except FileExistsError:
+            pass
+
+        return filepath_with_timestamp, "folder_created"
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            return "", "no_storage_space"
+        else:
+            logger.warning("An exception has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Storage_Available_Fault_Type, summary="Error while trying to create diagnostic logs folder")
+            diagnoser_output.append("An exception has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+            return "", "folder_not_created"
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Storage_Available_Fault_Type, summary="Error while trying to create diagnostic logs folder")
+        diagnoser_output.append("An exception has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+        return "", "folder_not_created"
+
+
+def connected_cluster_logger(filepath_with_timestamp, connected_cluster, storage_space_available):
```

Comment:
let's rename this :). Connected_cluster_logger might be misleading since it's not really a logger but we're capturing the GEt output. How about fetch_connected_cluster_resource() ?